stl/inc/algorithm

```diff
@@ -10394,12 +10403,28 @@ namespace ranges {
 
             using _Memcmp_classification_pred =
                 typename decltype(_Lex_compare_memcmp_classify(_First1, _First2, _Pred))::_Pred;
-            if constexpr (!is_void_v<_Memcmp_classification_pred> && sized_sentinel_for<_Se1, _It1> //
-                          && sized_sentinel_for<_Se2, _It2> && same_as<_Pj1, identity> && same_as<_Pj2, identity>) {
+            constexpr bool _Is_sized1 = sized_sentinel_for<_Se1, _It1>;
+            constexpr bool _Is_sized2 = sized_sentinel_for<_Se2, _It2>;
+            if constexpr (!is_void_v<_Memcmp_classification_pred> //
+                          && (_Is_sized1 || same_as<_Se1, unreachable_sentinel_t>) //
+#pragma warning(suppress : 6287) // Redundant code: the left and right subexpressions are identical
+                          &&(_Is_sized2 || same_as<_Se2, unreachable_sentinel_t>) //
+                          &&same_as<_Pj1, identity> && same_as<_Pj2, identity>) {

```

Comment:
In the above case you put those constraints on top. I would prefer to keep them there but downgrade to `is_same_v` or `_Same_impl`

Reply:
All of the ranges algorithm optimizations I've written already use `same_as`.